src/modules/rlm_redis_ippool/rlm_redis_ippool.c

```diff
@@ -295,16 +304,18 @@ static char lua_update_cmd[] =
 	 *	Update the expiry time
 	 */
 	"pool_key = '{' .. KEYS[1] .. '}:"IPPOOL_POOL_KEY"'" EOL			/* 14 */
-	"redis.call('ZADD', pool_key, 'XX', ARGV[1] + ARGV[2], ARGV[3])" EOL		/* 15 */
+	"local expires = tonumber(redis.call('ZSCORE', pool_key, ARGV[3]))" EOL
```

Comment:
What's going on here?  If the IP wasn't re-used, then we don't care about its previous ZSCORE?

Reply:
Oh, the staticness...

Reply:
Just change the scope of the first expires var?

Reply:
This is the only use of `expires` in this script